File: Day328_2025/Q328_Leetcode.py
from typing import List


class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        answer = []
        prefix = 0

        for bit in nums:
            prefix = ((prefix << 1) + bit) % 5
            answer.append(prefix == 0)

        return answer

File: Day328_2025/Q328_Leetcode.java
import java.util.ArrayList;
import java.util.List;

class Q328_Leetcode {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> answer = new ArrayList<>();
        int prefix = 0;

        for (int bit : nums) {
            prefix = ((prefix << 1) + bit) % 5;
            answer.add(prefix == 0);
        }

        return answer;
    }
}
